examples/layers/raw/spinning_square.dart

```diff
@@ -10,41 +12,60 @@ import 'dart:ui' as ui;
 Duration timeBase = null;
 
 void beginFrame(Duration timeStamp) {
-  Timeline.timeSync('beginFrame', () {
-    if (timeBase == null)
-      timeBase = timeStamp;
-    double delta = (timeStamp - timeBase).inMicroseconds / Duration.MICROSECONDS_PER_MILLISECOND;
-
-    // paint
-    ui.Rect paintBounds = ui.Point.origin & ui.window.size;
-    ui.PictureRecorder recorder = new ui.PictureRecorder();
-    ui.Canvas canvas = new ui.Canvas(recorder, paintBounds);
-    canvas.translate(paintBounds.width / 2.0, paintBounds.height / 2.0);
-    canvas.rotate(math.PI * delta / 1800);
-    canvas.drawRect(new ui.Rect.fromLTRB(-100.0, -100.0, 100.0, 100.0),
-                    new ui.Paint()..color = const ui.Color.fromARGB(255, 0, 255, 0));
-    ui.Picture picture = recorder.endRecording();
-
-    // composite
-    final double devicePixelRatio = ui.window.devicePixelRatio;
-    ui.Rect sceneBounds = new ui.Rect.fromLTWH(
-      0.0,
-      0.0,
-      ui.window.size.width * devicePixelRatio,
-      ui.window.size.height * devicePixelRatio
-    );
-    Float64List deviceTransform = new Float64List(16)
-      ..[0] = devicePixelRatio
-      ..[5] = devicePixelRatio
-      ..[10] = 1.0
-      ..[15] = 1.0;
-    ui.SceneBuilder sceneBuilder = new ui.SceneBuilder(sceneBounds)
-      ..pushTransform(deviceTransform)
-      ..addPicture(ui.Offset.zero, picture)
-      ..pop();
-    ui.window.render(sceneBuilder.build());
-  });
+  // The timeStamp argument to beginFrame indicates the timing information we
+  // should use to clock our animations. It's important to use timeStamp rather
+  // than reading the system time because we want all the parts of the system to
+  // coordinate the timings of their animations. If each component read the
+  // system clock independently, the animations that we processed later would be
+  // slightly ahead of the animations we processed earlier.
+
+  // The timeStamp given to us by the engine is based on some arbitrary zero. To
+  // make our animations reliable, we establish our own time base the first time
```

Comment:
it's not clear to me that this is necessary. we could just use timeStamp as the delta.